rllib/algorithms/algorithm.py

```diff
@@ -2690,11 +2732,17 @@ def _checkpoint_info_to_algorithm_state(
                     )
 
                 with open(policy_state_file, "rb") as f:
-                    worker_state["policy_states"][pid] = pickle.load(f)
+                    if msgpack is not None:
+                        worker_state["policy_states"][pid] = msgpack.load(f)
+                    else:
+                        worker_state["policy_states"][pid] = pickle.load(f)
 
             if policy_mapping_fn is not None:
                 worker_state["policy_mapping_fn"] = policy_mapping_fn
-            if policies_to_train is not None:
+            if (
+                policies_to_train is not None
+                or worker_state["is_policy_to_train"] == "__not_serializable__"
```

Comment:
can we add a comment / warning here to clarify that a policy is not being restored because the worker state is "__not_serializable__"

Reply:
my idea is that users should know explicitly that their policy isn't being restored for some reason

Reply:
Yeah, this needs commenting. There are only two things that I think we'll have problems with wrt checkpoints w/o code (msgpack):
- policy_mapping_function
- is_policy_to_train function

These are usually provided as lambdas and also might change quite a lot when users add and remove policies on the fly. This is why I completely left them out of the msgpack checkpoints. When users restore an algo from such a msgpack checkpoint, they need to provide these two functions again.

Reply:
Added better comments and improved error message for the case:
`Algorithm.from_checkpoint([msgpack checkpoint], policy_mapping_fn=None)`